### DISCUSSION TITLE
ScriptReference/Screen.SetResolution

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#297] [Notes]
# ScriptReference/Screen.SetResolution

https://docs.unity3d.com/ScriptReference/Screen.SetResolution.html

**Comment by joonjoonjoon** [comment]
Unity caches this, not in playerprefs somehow.  
I wrongly assumed that it would, reset to a default resolution, and was storing Screen.width as a base resolution before applying my own stored resolution.  
This is bad because Screen.width will return the unity cached value, not the monitor's native width.  
Use Screen.resolutions.Last() as a base instead.  
Observed in 2020.3.9f1
